### PR TITLE
restore &ipv6= tracker argument

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/http_tracker_connection.cpp [hunk 1]
```diff
@@ -204,6 +204,30 @@ namespace libtorrent
 					// source IP to determine our origin
 					url += "&ip=" + print_address(m_ses.listen_address());
 				}
+
+#if TORRENT_USE_IPV6
+				if (!tracker_req().ipv6 != address_v6() && !i2p)
```
Comment (updateing) [inline src/http_tracker_connection.cpp]: Is the the `!` before `tracker_req()` a typo? This line does not build with this `!` .

File: src/http_tracker_connection.cpp [hunk 2]
```diff
@@ -204,6 +204,30 @@ namespace libtorrent
...
+					std::string const ip = tracker_req().ipv6.to_string(err);
+					if (!err)
+					{
+						url += "&ipv6=";
+						url += ip;
+					}
+				}
+#endif
+
+				if (!tracker_req().ipv4 != address_v4() && !i2p)
```
Comment (updateing) [inline src/http_tracker_connection.cpp]: Same as above.

File: src/torrent.cpp [hunk 3]
```diff
@@ -2328,7 +2328,16 @@ namespace libtorrent
 		if (req.downloaded < 0) req.downloaded = 0;
 
 		req.event = e;
-		error_code ec;
+		if (!m_ses.m_settings.anonymous_mode)
+		{
+			tcp::endpoint ep;
+#if TORRENT_USE_IPV6
+			ep = m_ses.get_ipv6_interface();
+			if (ep != tcp::endpoint()) req.ipv6 = ep.address();
```
Comment (updateing) [inline src/torrent.cpp]: GCC complains about that `address` type can not be directly converted to `address_v6` nor `address_v4`.

**Comment by arvidn** [comment]
I thought I had updated the commit, but apparently I failed

**Comment by aldenml** [comment]
Is this change coming to RC_1_1? Also I see the code related to the `settings_pack::announce_double_nat` commented

**Comment by arvidn** [comment]
I will need to restrict this a bit. In general I don't think this feature was a good idea to begin with (which is why I removed it). I intend to remove the IPv4 argument altogether and only include the IPv6 argument for torrents marked "private".

Once this makes it into RC_1_0, I will forward port it into RC_1_1

**Comment by arvidn** [comment]
what about announce double nat? I don't actually recall what that does. is it passing in &ip=? (which also was never standard and also has the same security and privacy issues)

**Comment by aldenml** [comment]
Yes, for what I see in the code is to pass the `&ip=?` but it's commented (with a TODO) in RC_1_1. I'm not familiar with the feature, only asking for feature differences.

**Pull request merged**